src/layer.rs

```diff
@@ -948,25 +962,27 @@ where
             .get_mut::<OtelData>()
             .expect("Missing otel data span extensions");
 
-        let follows_span = ctx
-            .span(follows)
-            .expect("Span to follow not found, this is a bug");
-        let mut follows_extensions = follows_span.extensions_mut();
-        let follows_data = follows_extensions
-            .get_mut::<OtelData>()
-            .expect("Missing otel data span extensions");
-
-        let follows_context = self
-            .tracer
-            .sampled_context(follows_data)
-            .span()
-            .span_context()
-            .clone();
-        let follows_link = otel::Link::new(follows_context, Vec::new());
-        if let Some(ref mut links) = data.builder.links {
-            links.push(follows_link);
-        } else {
-            data.builder.links = Some(vec![follows_link]);
+        // The follows span may be filtered away from this layer, in which case
+        // we just drop the data, as opposed to panicking. This uses the same
+        // reasoning as `parent_context` above.
+        if let Some(follows_span) = ctx.span(follows) {
+            let mut follows_extensions = follows_span.extensions_mut();
+            let follows_data = follows_extensions
+                .get_mut::<OtelData>()
+                .expect("Missing otel data span extensions");
```

Comment:
Its unclear to me if this also needs to be `if let Some`...my understanding is that if `Context::span` returned span data for the follows span, then this lay did in fact see the follows span and added this data in `on_new_span`. This would mean that the `parent_context` function could also `.expect` getting the `OtelData` from the parent span, but Im not sure...

Reply:
I am not aware of cases where the contextual spans would not have their data in the registry so I think you are correct here that it's ok.